appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java

```diff
@@ -40,63 +54,107 @@ public static boolean hasWalletInstalled() {
-      return;
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+
+      createNotificationChannels("0", POA_NOTIFICATION_HEADS_UP, POA_NOTIFICATION_HEADS_UP,
+          notificationManager, NotificationManager.IMPORTANCE_HIGH);
+
+      createNotificationChannels("1", POA_NOTIFICATION_NORMAL, POA_NOTIFICATION_NORMAL,
+          notificationManager, NotificationManager.IMPORTANCE_DEFAULT);
+
+      Notification notificationHeadsUp = buildNotification("0", intent, true);
+      Notification notificationNormal = buildNotification("1", intent, false);
+
+      notificationManager.notify(1, notificationNormal);
+      notificationManager.notify(0, notificationHeadsUp);
+    } else {
+
+      Notification notificationHeadsUp = buildNotification("0", intent, true);
+
+      notificationManager.notify(0, notificationHeadsUp);
     }
+  }
 
-    dialogVisible = true;
-    AlertDialog.Builder builder;
-    builder = new AlertDialog.Builder(act);
-    builder.setTitle(R.string.wallet_missing);
-    builder.setMessage(act.getString(R.string.install_wallet_from_iab));
-    Log.d("String name: ", act.getString(R.string.install_wallet_from_iab));
-
-    builder.setPositiveButton(R.string.install, new DialogInterface.OnClickListener() {
-      @Override public void onClick(DialogInterface dialog, int which) {
-        act.startActivity(
-            new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName)));
-      }
-    });
+  private static void createNotificationChannels(String id, String name, String description,
+      NotificationManager notificationManager, int priority) {
+    NotificationChannel channelHeadUp = new NotificationChannel(id, name, priority);
+    channelHeadUp.setDescription(description);
+    notificationManager.createNotificationChannel(channelHeadUp);
+  }
 
-    builder.setNegativeButton(R.string.skip, new DialogInterface.OnClickListener() {
-      @Override public void onClick(DialogInterface dialogInterface, int i) {
-        dialogInterface.cancel();
-        dialogVisible = false;
-      }
-    });
+  private static Intent getNotificationIntent() {
+    Intent intent = null;
+    intent = new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName));
+    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
 
-    builder.setIcon(android.R.drawable.ic_dialog_alert);
-    builder.show();
+    if (hasAptoideInstalled()) {
+      intent.setPackage(aptoidePackageName);
+    }
+
+    PackageManager packageManager = context.getPackageManager();
+    ApplicationInfo applicationInfo = null;
+    Resources resources = null;
+
+    try {
+      applicationInfo =
+          packageManager.getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA);
+      resources = packageManager.getResourcesForApplication(applicationInfo);
+    } catch (PackageManager.NameNotFoundException e) {
+      e.printStackTrace();
+      return null;
```

Comment:
It this the desired effect? To not show the notification whent he application information is not available?